packages/storage/amplify_storage_s3/example/lib/main.dart

```diff
@@ -105,19 +108,271 @@ class HomeScreen extends StatefulWidget {
+    } on AuthException catch (e) {
+      _logger.debug('Could not check auth status - ${e.message}');
+    }
+  }
+
+  // upload a file to the S3 bucket
+  Future<void> _uploadFile() async {
+    final result = await FilePicker.platform.pickFiles(
+      type: FileType.custom,
+      allowedExtensions: ['jpg', 'png'],
+      withReadStream: true,
+      withData: false,
+    );
+
+    if (result == null) {
+      _logger.debug('No file selected');
+      return;
+    }
+
+    final platformFile = result.files.single;
+
+    try {
+      final result = await Amplify.Storage.uploadFile(
+        localFile: AWSFile.fromStream(
+          platformFile.readStream!,
+          size: platformFile.size,
+        ),
+        key: platformFile.name,
+        onProgress: (p) =>
+            _logger.debug('Uploading: ${p.transferredBytes}/${p.totalBytes}'),
+      ).result;
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.debug('Error uploading file - ${e.message}');
+    }
+  }
+
+  // list all files in the S3 bucket
+  Future<void> _listAllPublicFiles() async {
+    try {
+      final result = await Amplify.Storage.list(
+        options: const StorageListOptions(
+          accessLevel: StorageAccessLevel.guest,
+          pluginOptions: S3ListPluginOptions.listAll(),
+        ),
+      ).result;
+      setState(() {
+        list = result.items;
+      });
+    } on StorageException catch (e) {
+      _logger.debug('List error - ${e.message}');
+    }
+  }
+
+  // download file on mobile
+  Future<void> downloadFileMobile(String key) async {
+    final documentsDir = await getApplicationDocumentsDirectory();
+    final filepath = '${documentsDir.path}/$key';
+    try {
+      final result = await Amplify.Storage.downloadFile(
+        key: key,
+        localFile: AWSFile.fromPath(filepath),
+        onProgress: (p0) => _logger
+            .debug('Progress: ${p0.transferredBytes} / ${p0.totalBytes}%)'),
+      ).result;
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.debug('Download error - ${e.message}');
+    }
+  }
+
+  // download file on web
+  Future<void> downloadFileWeb(String key) async {
+    try {
+      final result = await Amplify.Storage.downloadFile(
+        key: key,
+        localFile: AWSFile.fromPath(key),
+      ).result;
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.debug('Download error - ${e.message}');
+    }
+  }
+
+  // delete file from S3 bucket
+  Future<void> removeFile({
+    required String key,
+    required StorageAccessLevel accessLevel,
+  }) async {
+    try {
+      final result = await Amplify.Storage.remove(
+        key: key,
+        options: StorageRemoveOptions(accessLevel: accessLevel),
+      ).result;
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.debug('Delete error - ${e.message}');
+      rethrow;
```

Comment:
Any reason we are throwing here instead of just logging?

Throwing here will result in an unhandled exception in the Flutter runtime which will present an ugly red screen when running in debug mode. Better to either just log or show some visual indication like a [SnackBar](https://api.flutter.dev/flutter/material/SnackBar-class.html).